### PR TITLE
Adding per API log feature's and log4j2 UI manipulation docs

### PR DESCRIPTION
**Pull request by dinusha92**
With a particular WUM update 1600491803101, these two features got shipped. 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/administer/logging-and-monitoring/logging/gw-api.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          description: |
+            context and the logLevel can be provided as a payload. This is useful if you want to enable logs for multiple APIs at once.
+          required: false
+          in: body
+          schema:
+            $ref: '#/definitions/APIList'
+        - name: context
+          in: query
+          description: |
+            API context ( context/version eg: pizzashack/1.0.0)
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/gw-api.yaml:36]: if you give context/version of the API, how do you enable it for multiple APIs? shall we add an example?

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+template: templates/swagger.html
+---
+# Setting up logging per API in API Manager
+
+!!! note
+    Enabling logs per API is only available after the WUM update with the timestamp "1600491803101"
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:7]: we usually give out the date the WUM was released. Can we add that info?

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+In a production envrionment, having the feasbility to monitor HTTP message flow for a given API is important in identifying issues in the HTTP message flow.
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:9]: HTTP message flow -> the HTTP message flow

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+This feature enables API requests logging at four levels.
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:11]: API request logging -> logging API requests

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+
+1. Request coming to the gateway from client (request-in>>)
+2. Request leaving the gateway to backend (request-out<<)
+3. Request coming to the gateway from backend (response-in>>)
+4. Request leaving to the client from gateway (response-out<<)
+
+Depending on the nature of the request, you can chose only to log headers, body or both.
+
+A REST API secured with basic authentication is provided to enable/disable API logs at the run time. Only a user with admin role has priviledges to invoke the API.
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:21]: Only a user with admin privileges is able to invoke the API

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:23]: remove empty lines

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+??? info "Click here to see how to try out the APIs"
+    Do the following to try out the REST APIs with your local instance of WSO2 API Manager.
+
+       1. Expand the relevant API operation and click the **Try It Out** button.
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:29]: We need to provide steps to find this, to guide a user from the UI.

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md [hunk 8]
```diff
@@ -66,6 +66,29 @@ Please follow below steps to enable logs for a given service component available
     loggers = org-wso2-carbon-user-core, trace-messages, org-apache-coyote,com-hazelcast
     ```
 
+###Enable Logs for a Component via UI
+
+!!! note
+    Enabling logs via UI is only available after the WUM update with the timestamp "1600491803101"
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md:72]: same as above.

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md [hunk 9]
```diff
@@ -66,6 +66,29 @@ Please follow below steps to enable logs for a given service component available
...
+
+Please follow below steps to enable logs for a given service component available in WSO2 API Manager.
+
+1.  Navigate to the carbon console [https://localhost:9443/carbon](https://localhost:9443/carbon) and sign in with **`admin/admin`** as the credentials.
+Go to "Configure" -> "Logging"
+                                                 
+    ![Logger Carbon UI](../../../assets/img/administer/logging.png)
```
Comment (shaniR) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md:79]: please use {{base_path}}

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/administer/logging-and-monitoring/logging/gw-api.yaml (hunk 1 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/gw-api.yaml:36]: This is already shown once you expand the swagger console.Reference scheme has been provided.

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 2 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md]: Fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 3 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md]: fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 4 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md]: fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 5 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md]: fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 6 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md]: fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md (hunk 7 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging-per-api.md:26]: Swagger UI is there to populate the documentation from the given swagger definition.

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md (hunk 8 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md]: fixed

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md (hunk 9 above)
Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md]: The local setup did not show the images when the basePath was provided. Could be an issue with local setup. But now I modified using the basepath.

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md [hunk 10]
```diff
@@ -1,74 +1,105 @@
...
+WSO2 API Manager is shipped with the log4j2 logging capabilities that generate logs for administrative and server-side activities. By default, Carbon Logs are persisted in the `wso2carbon.log` file, which is located in thee `<APIM_HOME>/repository/logs` directory. You can configure the details that are captured in this log file by configuring the `<APIM_HOME>/repository/conf/log4j2.properties` file.
 
-!!!Java logging and Log4j integration
+!!! text "Java Logging and Log4j integration"
     In addition to the logs from libraries that use Log4j, all logs from libraries such as, Tomcat and Hazelcast that use Java logging framework are also visible in the same log files. That is, when Java logging is enabled in Carbon, only the Log4j appenders will write to the log files. If the Java Logging Handlers have logs, these logs will be delegated to the log events of the corresponding Log4j appenders. A Pub/Sub registry pattern implementation has been used in the latter mentioned scenario to plug the handlers and appenders.
```
Comment (Mariangela) [inline en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md:12]: @dinusha92 , is it ok to mention Hazelcast here in the note and in the configurations?

Comment (dinusha92) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md:12]: @Mariangela this has been there for a time. Actually I am not aware why this has been added initially.

**Review by shaniR — APPROVED** [review]

File: en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md (hunk 10 above)
Comment (Mariangela) [reply en/docs/administer/logging-and-monitoring/logging/setting-up-logging.md:12]: @ruks , please comment whether it is correct to state Hazelcast here

**Pull request merged**